docs/query-builder.txt

```diff
@@ -9,588 +9,1116 @@ Query Builder
+   run similar database operations. To learn more about retrieving documents
+   by using Eloquent models, see :ref:`laravel-fundamentals-retrieve`.
 
-When using ``whereNotIn`` objects will be returned if the field is
-non-existent. Combine with ``whereNotNull('age')`` to omit those documents.
+Laravel provides a **facade** to access the query builder class ``DB``, which
+lets you perform database operations. Facades, which are static interfaces to
+classes, make the syntax more concise, avoid runtime errors, and improve
+testability.
 
-**whereBetween**
+{+odm-short+} aliases the ``DB`` method ``table()`` as the ``collection()``
+method. Chain methods to specify commands and any constraints. Then, chain
+the ``get()`` method at the end to run the methods and retrieve the results.
+The following example shows the syntax of a query builder call:
 
 .. code-block:: php
 
-   $posts = Post::whereBetween('votes', [1, 100])->get();
+   DB::collection('<collection name>')
+       // chain methods by using the "->" object operator
+       ->get();
 
-**whereNull**
+This guide provides examples of the following types of query builder operations:
+
+- :ref:`laravel-retrieve-query-builder`
+- :ref:`laravel-modify-results-query-builder`
+- :ref:`laravel-mongodb-read-query-builder`
+- :ref:`laravel-mongodb-write-query-builder`
+
+Before You Get Started
+----------------------
+
+To run the code examples in this guide, complete the
+:ref:`Quick Start <laravel-quick-start>` tutorial to configure a web
+application, load sample datasets into your MongoDB deployment, and
+run the example code from a controller method.
+
+To perform read and write operations by using the query builder, import the
+``Illuminate\Support\Facades\DB`` facade and compose your query.
+
+.. _laravel-retrieve-query-builder:
+
+Retrieve Matching Documents
+---------------------------
+
+This section includes query builder examples for read operations in the
+following operator categories:
+
+- :ref:`Where method <laravel-query-builder-where-example>`
+- :ref:`Logical conditionals <laravel-query-builder-logical-operations>`
+- :ref:`Ranges and type checks <laravel-query-builder-range-type>`
+- :ref:`Pattern searches <laravel-query-builder-pattern>`
+- :ref:`Retrieve distinct values <laravel-query-builder-distinct>`
+- :ref:`Aggregations <laravel-query-builder-aggregations>`
+
+.. _laravel-query-builder-where-example:
+
+Where Method Example
+~~~~~~~~~~~~~~~~~~~~
+
+The following example shows how to use the ``where()`` query
+builder method to retrieve documents from the ``movies`` collection
+that contain an ``imdb.votes`` field value of ``350``. Click the
+:guilabel:`{+code-output-label+}` button to see the results returned
+by the query:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/query-builder/QueryBuilderTest.php
+      :language: php
+      :dedent:
+      :start-after: begin query where
+      :end-before: end query where
```

Comment:
I: description does not match example
imdb.votes field value of 350
where('imdb.rating', 9.3)